marklogic-data-hub/src/main/java/com/marklogic/hub/flow/impl/FlowRunnerImpl.java

```diff
@@ -103,15 +104,15 @@ public RunFlowResponse runFlow(String flowName, List<String> stepNums, String jo
             flow.setOverrideDestDB(destDB);
         }
 
-        if(sourceDB == null){
```

Comment:
Please use if (StringUtils.isNotEmpty(sourceDB)) {..} the same for other checks.
https://github.com/marklogic/marklogic-data-hub/blob/9fdff17adb1ec88e052ce5de153691fdc70bf01c/marklogic-data-hub/src/main/java/com/marklogic/hub/flow/impl/FlowRunnerImpl.java#L119

        if (options != null && !options.isEmpty()) {
            flow.setOverrideOptions(options);
        }

Reply:
1. options is a map and can be empty, so no need for the check
2. will use StringUtils.isNotEmpty for sourceDB

Reply:
if the map is empty, do you still want to ovveride the Options in Flow?

Reply:
It is not going to affect anything as they will be passed to stepRunner,  which can add its options( if present)  and will be sent to the server where they are combined.